pkg/repository/v1manifest/repo.go

```diff
@@ -58,7 +82,72 @@ func Init(dst string, initTime time.Time) error {
 		return fmt.Errorf("manifest '%s' not initialized porperly", ty)
 	}
 
-	return BatchSaveManifests(dst, manifests)
+	return BatchSaveManifests(dst, manifests, privKey)
+}
+
+// AddComponent adds a new component to an existing repository
+func AddComponent(id, name, desc, owner, repo string, isDefault bool, pub, priv string) error {
+	id = strings.ToLower(id)
```

Comment:
Don't do this, the id should be taken literally and can include case differences